fili-core/src/test/groovy/com/yahoo/bard/webservice/web/FiliResponseWriterSpec.groovy

```diff
@@ -0,0 +1,167 @@
+        filiResponseWriterSelector = new FiliResponseWriterSelector(csvResponseWriter, jsonResponseWriter, jsonApiResponseWriter);
+        filiResponseWriter = new FiliResponseWriter(filiResponseWriterSelector)
+        filiResponseWriter.write(apiRequest, paginatedResponse, os)
+
+        expect:
+        GroovyTestUtils.compareJson(os.toString(), expectedJson)
+
+        where:
+        linkNames << LINK_NAMES_LIST
+        bodyLinks << BODY_LINKS_LIST
+        bodyLinksAsJson << BODY_LINKS_AS_JSON_LIST
+    }
+
+    @Unroll
+    def "test for existence of missing intervals in response when #arePaginating"() {
+        setup:
+        boolean partialDataSetting = PARTIAL_DATA.isOn()
+        PARTIAL_DATA.setOn(true)
+
+        formattedDateTime = dateTime.toString(getDefaultFormat())
+        GString paginationBlock = """, "pagination": {
+                        $bodyLinksAsJson,
+                        "currentPage": $PAGE,
+                        "rowsPerPage": $PER_PAGE,
+                        "numberOfResults": 6
+                    }"""
+        GString metaBlock = """{
+                "missingIntervals" : [
+                    "2014-07-01 00:00:00.000/2014-07-08 00:00:00.000",
+                    "2014-07-15 00:00:00.000/2014-07-22 00:00:00.000"
+                 ]
+                 ${paginating ? paginationBlock : ""}
+                  }
+                }"""
+
+        SimplifiedIntervalList missingIntervals = [new Interval("2014-07-01/2014-07-08"), new Interval(
+                "2014-07-15/2014-07-22"
+        )] as SimplifiedIntervalList
+        ResponseData response1 = new ResponseData(
+                resultSet,
+                apiRequest,
+                missingIntervals,
+                volatileIntervals,
+                paginating ? pagination : null,
+                bodyLinks
+        )
+
+        String expectedJson = withMetaObject(defaultJsonFormat, metaBlock)
+
+        csvResponseWriter = new CsvResponseWriter(MAPPERS)
+        jsonResponseWriter = new JsonResponseWriter(MAPPERS)
+        jsonApiResponseWriter = new JsonApiResponseWriter(MAPPERS)
+        filiResponseWriterSelector = new FiliResponseWriterSelector(csvResponseWriter, jsonResponseWriter, jsonApiResponseWriter);
+        filiResponseWriter = new FiliResponseWriter(filiResponseWriterSelector)
+        filiResponseWriter.write(apiRequest, response1, os)
+
+        expect:
+        GroovyTestUtils.compareJson(os.toString(), expectedJson)
+
+        cleanup:
+        PARTIAL_DATA.setOn(partialDataSetting)
+
+        where:
+        paginating << [false] + [true] * LINK_NAMES_LIST.size()
+        linkNames << [''] + LINK_NAMES_LIST
+        bodyLinks << [[:] as Map] + BODY_LINKS_LIST
+        bodyLinksAsJson << [''] + BODY_LINKS_AS_JSON_LIST
+        arePaginating = paginating ? "paginating with pagination links $linkNames" : "not paginating"
+    }
+
+
+
+    def "Test dimension column name builds correctly"() {
```

Comment:
This is a test on a method in `ResponseData`, not in `FiliResponseWriter`, so it should be in the `ResponseDataSpec`, not the `FiliResponseWriterSpec`.